src/tools/qt-gui/src/confignode.cpp

```diff
@@ -11,9 +11,9 @@
 
 using namespace kdb;
 
-ConfigNode::ConfigNode (QString name, QString path, const Key & key, TreeViewModel * parentModel)
+ConfigNode::ConfigNode (QString name, QString path, const Key & key, TreeViewModel * parentModel, bool root)
 : m_name (std::move (name)), m_path (std::move (path)), m_key (key), m_children (new TreeViewModel), m_metaData (nullptr),
-  m_parentModel (parentModel), m_isExpanded (false), m_isDirty (false)
+  m_parentModel (parentModel), m_isExpanded (false), m_isDirty (false), m_root (root)
 {
```

Comment:
If it stays this way you would have to add the 'm_root(root)' part to the copy constructor too?

Reply:
m_isRoot (other.m_isRoot)